lib/classes/Variables.test.js

```diff
@@ -156,6 +158,30 @@ describe('Variables', () => {
             .then(() => expect(property.getter(awsProvider)).to.be.eql('default'));
         });
       });
+
+      it('should not populate variables containing *', () => {
+        const property = { name: 'region', getter: (provider) => provider.getRegion() };
+
+        awsProvider.options[property.name] = '${self:foobar, *.a.thing}';
+        serverless.variables.populateService().should.be.fulfilled
+          .then(() => expect(property.getter(awsProvider)).to.be.eql('${self:foobar, *.a.thing}'));
```

Comment:
So you can disable variable resolution by accidentally typing in an asterisk?  That may be fine but could confuse without offering warning.

Reply:
Good point.

The existing behavior is that variable resolution will be disabled if an `*` is in the declaration, regardless of `"` `'`.

It probably does make sense to add some warning.

Are there any existing PR's which would serve as a good template for adding such a warning?

Reply:
Sorry, context loss error on my part.  As existing behavior, I would not expect you to fix it.  Your changes reduce the surface space of the unexpected/odd/confusion behavior and that's an improvement over current state.